### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by HStakhiv**

### CODE REVIEW

File: cinema/views.py [hunk 1]
```diff
@@ -1 +1,71 @@
...
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action in ("list", "retrieve"):
+            queryset = queryset.prefetch_related("movie", "cinema_hall")
```
Comment (VadymShkarbul) [inline cinema/views.py:60]: Maybe you should use select_related here (but django-debug-toolbar says that no matter)

**Review by VadymShkarbul — APPROVED** [review]

File: cinema/views.py (hunk 1 above)
Comment (HStakhiv) [reply cinema/views.py:60]: Thanks. Yes

File: cinema/models.py [hunk 2]
```diff
@@ -27,6 +27,10 @@ class Actor(models.Model):
     first_name = models.CharField(max_length=255)
     last_name = models.CharField(max_length=255)
 
+    @property
+    def full_name(self):
+        return f"{self.first_name} {self.last_name}"
+
     def __str__(self):
         return self.first_name + " " + self.last_name
```
Comment (Nattalli) [inline cinema/models.py:35]: 1) add a type annotation
2) These two methods do the same thing. Maybe it's better to use one of them in the other?

**Review by Nattalli — CHANGES_REQUESTED** [review]

File: cinema/models.py (hunk 2 above)
Comment (HStakhiv) [reply cinema/models.py:35]: Sorry. I do not agree with point number 2. The two methods are the same, but if I leave, for example, only the str method and change it in the future, then I will lose the functionality of the full_name field.  I want full_name to be independent, so I created this property.

Comment (HStakhiv) [reply cinema/models.py:35]: Sorry. I realized. I'll just do `return full_name()` in `str`
Thanks

Comment (Nattalli) [reply cinema/models.py]: btw, I partially agree about independence. But if you leave these methods independent, then they should at least be written in the same way inside (adhere to the same style)

**Review by Nattalli — APPROVED** [review]

**Pull request closed without merging**